sdk/src/pubkey.rs

```diff
@@ -54,11 +66,31 @@ impl fmt::Display for Pubkey {
     }
 }
 
+pub fn write_pubkey(outfile: String, pubkey: Pubkey) -> Result<(), Box<error::Error>> {
+    let printable = format!("{}", pubkey);
+    let serialized = serde_json::to_string(&printable)?;
+
+    if let Some(outdir) = Path::new(&outfile).parent() {
+        fs::create_dir_all(outdir)?;
+    }
+    let mut f = File::create(outfile)?;
+    f.write_all(&serialized.clone().into_bytes())?;
+
+    Ok(())
+}
+
+pub fn read_pubkey(infile: &str) -> Result<Pubkey, Box<error::Error>> {
```

Comment:
I'd prefer we use the `Box<error::Error>` trick as rarely as possible. Maybe for a top-level CLI function, but hopefully never in libraries. We want to tell the caller the exact set of possible errors they might need to handle. Using this trick tells them the function may return any error. That's not too helpful.

Reply:
Hmm. There are 3 different error types that this method could encounter. It seems like overkill to invent a new one to map them to. How would you suggest I handle them instead?

Presumably, you's also like me to rewrite the similar signature methods? https://github.com/solana-labs/solana/blob/1d48c4dd45b233c5406d853afaef4f1b2d4a26c9/sdk/src/signature.rs#L93